### PR TITLE
feat(poe-v2): support poe v2 on trends

### PR DESCRIPTION
**Pull request by yakkomajuri**
Support for PoE v2 on the whole trends stack: trends, lifecycle, breakdowns.

Currently only a couple of tests for the new functionality, but provided the existing ones pass we can be sure the current functionality won't break.

### CODE REVIEW

File: posthog/queries/test/__snapshots__/test_trends.ambr [hunk 1]
```diff
@@ -3,7 +3,7 @@
   
   SELECT groupArray(value)
   FROM
-    (SELECT replaceRegexpAll(JSONExtractRaw(group0_properties, 'industry'), '^"|"$', '') AS value,
+    (SELECT replaceRegexpAll(JSONExtractRaw(group_properties_0, 'industry'), '^"|"$', '') AS value,
```
Comment (tiina303) [inline posthog/queries/test/__snapshots__/test_trends.ambr:6]: Do I understand correctly that we're using latest group properties here?

File: posthog/queries/test/__snapshots__/test_trends.ambr [hunk 2]
```diff
@@ -17,7 +17,6 @@
        AND event = 'sign up'
        AND toTimeZone(timestamp, 'UTC') >= toDateTime('2020-01-01 00:00:00', 'UTC')
        AND toTimeZone(timestamp, 'UTC') <= toDateTime('2020-01-12 23:59:59', 'UTC')
-       AND notEmpty(e.person_id)
```
Comment (tiina303) [inline posthog/queries/test/__snapshots__/test_trends.ambr:20]: not sure if this is now verified somewhere else or not?

**Review by tiina303 — COMMENTED** [review]
Why so many snapshot  changes - I would expect just to see the two added tests, I have no confidence about those changes being right and I don't know where they are coming from

**Comment by mariusandra** [comment]
The snapshot changes happen because in CI we run our tests multiple times -> with PoE on and PoE off. When the "make a commit if snapshot changes" bot runs, you never know what ends up in the repo: with PoE on or off. It's usually stable, but not always. The same thing seems to have happened here -> PoE on turned into PoE off and vice versa.

The correct fix is to redo the CI pipeline to not run each test twice. We'll get there... 😅

**Comment by neilkakkar** [comment]
Pretty sure snapshot updates used to run only on non-person-on-events tests (unless there's a test that manually sets this up): https://github.com/PostHog/posthog/blob/master/.github/workflows/ci-backend.yml#L267 - so if that's not happening, something borked this 😅

**Comment by neilkakkar** [comment]
Ah, the borking here might be that p-o-e is running for tests its not supposed to (i.e. when the env var isn't set). If this assumption has changed, worth updating the CI too.

**Comment by yakkomajuri** [comment]
Some weird unrelated test failures that pass locally like:

```
posthog/api/test/test_authentication.py::TestPasswordResetAPI::test_user_can_reset_password
```

**Comment by yakkomajuri** [comment]
FOSS tests just probably shouldn't run with PoE whereas EE tests are weird because they pass locally... Investigating

File: ee/clickhouse/queries/test/__snapshots__/test_retention.ambr [hunk 3]
```diff
@@ -364,15 +376,8 @@
           NULL as selected_interval,
           returning_event_query as
        (SELECT toStartOfWeek(toTimeZone(toDateTime(e.timestamp, 'UTC'), 'UTC')) AS event_date,
-               pdi.person_id as target
+               e.person_id as target
         FROM events e
-        INNER JOIN
```
Comment (tiina303) [inline ee/clickhouse/queries/test/__snapshots__/test_retention.ambr:369]: should this be changing?

**Review by tiina303 — APPROVED** [review]

File: posthog/queries/trends/breakdown.py [hunk 4]
```diff
@@ -233,9 +240,6 @@ def get_query(self) -> Tuple[str, Dict, Callable]:
                     person_join=person_join_condition,
                     groups_join=groups_join_condition,
                     sessions_join=sessions_join_condition,
-                    person_id_alias=self.DISTINCT_ID_TABLE_ALIAS
```
[the remark this answers is not in the document]
Comment (yakkomajuri) [reply posthog/queries/trends/breakdown.py:236]: hmm seems I just deleted this? good spot

[the remark this answers is not in the document]
Comment (yakkomajuri) [reply posthog/queries/trends/breakdown.py:236]: ah it was just unused on this query

File: ee/clickhouse/queries/test/__snapshots__/test_retention.ambr (hunk 3 above)
Comment (yakkomajuri) [reply ee/clickhouse/queries/test/__snapshots__/test_retention.ambr:369]: Actually, this particular one yes. Somehow this was wrong on master? Note how it now correctly uses PoE v1 and it is a `person_on_events` test

File: ee/clickhouse/queries/test/__snapshots__/test_paths.ambr [hunk 5]
```diff
@@ -1102,18 +1096,11 @@
                      groupArray(path_item) as paths
               FROM
                 (SELECT e.timestamp AS timestamp,
-                        pdi.person_id AS person_id,
+                        e.person_id AS person_id,
                         if(e.event = '$screen', replaceRegexpAll(JSONExtractRaw(properties, '$screen_name'), '^"|"$', ''), if(e.event = '$pageview', if(length(replaceRegexpAll(JSONExtractRaw(properties, '$current_url'), '^"|"$', '')) > 1, replaceRegexpAll(replaceRegexpAll(JSONExtractRaw(properties, '$current_url'), '^"|"$', ''), '/$', ''), replaceRegexpAll(JSONExtractRaw(properties, '$current_url'), '^"|"$', '')), e.event)) AS path_item_ungrouped,
                         multiMatchAnyIndex(path_item_ungrouped, NULL) AS group_index,
                         if(group_index > 0, NULL[group_index], path_item_ungrouped) AS path_item
                  FROM events e
-                 INNER JOIN
```
[the remark this answers is not in the document]
Comment (yakkomajuri) [reply ee/clickhouse/queries/test/__snapshots__/test_paths.ambr:1110]: Looked into - yes it should. Note how it is a `person_on_events` test. Must've been wrong on master. 

Really really appreciate you looking at these diffs though

**Pull request merged**